akka-stream/src/main/scala/akka/stream/impl/io/TcpConnectionStream.scala

```diff
@@ -91,7 +89,6 @@ private[akka] abstract class TcpStreamActor(val settings: ActorMaterializerSetti
       if (!closed) {
         closed = true
         pendingElement = null
-        if (!tcpOutputs.isFlushed && (connection ne null)) connection ! ResumeReading
```

Comment:
this will stop draining from the input, which has the side-effect of potentially missing close events. The idea is that cancellation turns the TCP read side to automatic drain-and-discard to handle these cases.